src/main/scala/viper/silver/ast/Ast.scala

```diff
@@ -48,7 +49,7 @@ Some design choices:
   * Note that all but Program are transitive subtypes of `Node` via `Hashable`. The reason is
   * that AST node hashes may depend on the entire program, not just their sub-AST.
   */
-trait Node extends Traversable[Node] with Rewritable {
+trait Node extends Iterable[Node] with Rewritable {
 
   /** @see [[Nodes.subnodes()]] */
   def subnodes = Nodes.subnodes(this)
```

Comment:
@fpoli If we bake in the assumption that all nodes are immutable, we could turn `subnodes` and `iterator` (and maybe other `def`s) into `lazy val`s. This could speed up subsequent traversals of the same tree.